### PR TITLE
build: update `@angular/bazel` to latest

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

File: src/bazel-tsconfig-build.json [hunk 1]
```diff
@@ -21,8 +21,6 @@
     "importHelpers": true,
     "strictBindCallApply": true,
     "esModuleInterop": true,
-    // Keep the below in sync with https://github.com/angular/angular/blob/f9b821f07d8dba57a6a7e5fc127dc096247424aa/packages/bazel/src/ng_module/ng_module.bzl#L214
-    "useDefineForClassFields": false,
```
Comment (crisbeto) [inline src/bazel-tsconfig-build.json:25]: Does this mean that we now have `useDefineForClassFields` enabled?

Comment (alan-agius4) [reply src/bazel-tsconfig-build.json:25]: Yes

Comment (crisbeto) [reply src/bazel-tsconfig-build.json:25]: Is there a reason for this? I remember it had an impact on the bundle size.

Comment (alan-agius4) [reply src/bazel-tsconfig-build.json:25]: It did increase the bundle size of on older Safari versions that are no longer supported. 

However, this option does improve DCE in general.

**Review by crisbeto — APPROVED** [review]

**Pull request merged**